spec/importc.dd

```diff
@@ -297,8 +297,21 @@ $(H2 $(LNAME2 other-solutions, Other Solutions))
 
 $(H2 $(LNAME2 internals, How ImportC Works))
 
-    $(P This is a description of how ImportC is implemented, intended
-    to remove the mystery of various design choices.)
+    $(P ImportC's implementation is based on the idea that D's semantics are very similar
+    to C's. ImportC gets its own parser, which converts the C syntax into the same AST
+    (Abstract Syntax Tree) that D uses. The lexer for ImportC is the same as for D, but with
+    some modifications here and there, such as the keywords and integer literals being different.
+    Where the semantics of C differ from D, there are adjustments in the semantic analysis
+    code in the D compiler.)
+
+    $(P This co-opting of the D semantic implementation allows ImportC to be able to do things
+    like handle forward references, CTFE (Compile Time Function Execution), and inlining of C functions
+    into D code. Being able to handle forward references means it is not necessary to even
+    write a .h file to be able to import C declarations into D. Being able to perform CTFE is
+    very handy for testing that ImportC is working without needing to generated an executable.
+    But, in general, the strong temptation to add D features to ImportC has been resisted.)
```

Comment:
What "strong temptation to add D features"? I thought it was very clear from the start that the goal was to import existing C code, not write new C with D features mixed in.

Reply:
That is clear. But I am tempted, and people will ask "why not" add some.

Reply:
What kind of D features are you tempted to add to ImportC? I think this paragraph leaves some open questions to the reader, another example:

> Being able to perform CTFE is very handy for testing that ImportC is working without needing to generate an executable.

Why is this handy?

Reply:
> What kind of D features are you tempted to add to ImportC?

Like make . work as a substitute for ->

> Why is this handy?

The test suite runs much faster if it doesn't have to generate an executable.